internal/pipe/webhook/webhook.go

```diff
@@ -0,0 +1,158 @@
+package webhook
+
+import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/apex/log"
+	"github.com/caarlos0/env/v6"
+	"github.com/goreleaser/goreleaser/internal/tmpl"
+	"github.com/goreleaser/goreleaser/pkg/context"
+)
+
+const (
+	defaultMessageTemplate = `{ "message": "{{ .ProjectName }} {{ .Tag }} is out! Check it out at {{ .ReleaseURL }}"}`
+	ContentTypeHeaderKey   = "Content-Type"
+	UserAgentHeaderKey     = "User-Agent"
+	UserAgentHeaderValue   = "gorleaser"
+	AuthorizationHeaderKey = "Authorization"
+	DefaultContentType     = "application/json; charset=utf-8"
+
+	EndpointRegex = "(http|nats)(s?)://.*"
+)
+
+var (
+	// ErrHeaderMissing = 400
+	ErrHeaderMissing = errors.New("header missing")
+	// ErrClientAuthenticationError = 401
+	ErrClientAuthenticationError = errors.New("authentication Error")
+	// ErrForbidden = 403
+	ErrForbidden = errors.New("access Denied")
+	// ErrNotFound = 404
+	ErrNotFound = errors.New("resource not found")
+	// ErrUnprocessableEntityError = 422
+	ErrUnprocessableEntityError = errors.New("bad Request")
+	// ErrTooManyRequest = 429
+	ErrTooManyRequest = errors.New("exceeding post rate limit")
+)
+
+type Pipe struct{}
+
+func (Pipe) String() string                 { return "webhook" }
+func (Pipe) Skip(ctx *context.Context) bool { return !ctx.Config.Announce.Webhook.Enabled }
+
+type Config struct {
+	BasicAuthHeader   string `env:"BASIC_AUTH_HEADER_VALUE"`
+	BearerTokenHeader string `env:"BEARER_TOKEN_HEADER_VALUE"`
+}
+
+func (p Pipe) Default(ctx *context.Context) error {
+	if ctx.Config.Announce.Webhook.MessageTemplate == "" {
+		ctx.Config.Announce.Webhook.MessageTemplate = defaultMessageTemplate
+	}
+	if ctx.Config.Announce.Webhook.ContentType == "" {
+		ctx.Config.Announce.Webhook.ContentType = DefaultContentType
+	}
+	if !ctx.Config.Announce.Webhook.CheckCert {
+		ctx.Config.Announce.Webhook.CheckCert = true
+	}
```

Comment:
since `CheckCert` is a bool, this means it will always check cert...

maybe another approach would be to use something like `skip_tls_verify` (or something along these lines), which defaults to `false`...

Reply:
Good point, i changed it! Thx for pointing out!